proposed/2021/TargetFrameworkAndPlatformsFilters.md

```diff
@@ -0,0 +1,196 @@
+
+- User selects `Any` for the .NET generation.
+  - Version filter populates all known versions for .NET Framework, .NET Core, .NET, and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
+- User selects `Any` for the .NET generation and `Windows` for the supported platform.
+  - Version filter populates all known version for .NET Framework, .NET Core, .NET, and .NET Standard.
+  - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
+- User selects `Any` for the .NET generation and `Android` for the supported platform.
+  - Version filter populates all known version for .NET and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 6.0
+- User selects `.NET Framework` for the .NET generation.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8
+- User selects `.NET Framework` for the .NET generation and `Windows` for the supported platform.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8
+- User selects `.NET Core` for the .NET generation and `macOS` for the supported platform.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1
+
+For further scenarios, see the [framework precedence](https://nugettools.azurewebsites.net/6.0.0-preview.3/framework-precedence) tool to understand how the version matrix would be calculated.
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/VersionNumbers.png)
+
+#### Complete Experience
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/CompleteExperience.png)
+
+## Search Syntax
+
+Lastly, NuGet should add a protocol in which supports [search syntax](https://docs.microsoft.com/nuget/consume-packages/finding-and-choosing-packages#search-syntax) for frameworks, platforms, and potentially *implementations**.
+
+Here's an ideal `<property>:<term>` for each use-case:
+
+```
+framework:net472 framework:net471 framework:net47   # Match the target framework monikers
+
+platform:windows platform:macos platform:linux    # Match the platforms
+
+implementation:netframework implementation:netcore implementation:net  # Match the generation of .NET including both supported frameworks & platforms.
+```
+
+- `framework:` - Allows a short-form TFM value to serach on.
+- `platform:` - Allows a platform value to search on.
+- `implementation` - Allows a generation of .NET to search on (includes all frameworks of the generaiton).
```

Comment:
```suggestion
- `implementation:` - Allows a generation of .NET to search on (includes all frameworks of the generation).
```

Reply:
Are the `platform:` and `implementation:` filters necessary? Can we power the MVP using only the `framework:` filter?

Reply:
I would say `platform:` is a necessary filter. The other not so much other than being a helpful shorthand.